File: scalardb-sql-jdbc-sample/src/main/java/sample/command/GetCustomerInfoCommand.java
package sample.command;

import java.util.concurrent.Callable;
import picocli.CommandLine.Command;
import picocli.CommandLine.Parameters;
import sample.Sample;

@Command(name = "GetCustomerInfo", description = "Get customer information")
public class GetCustomerInfoCommand implements Callable<Integer> {

  @Parameters(index = "0", paramLabel = "CUSTOMER_ID", description = "customer ID")
  private int customerId;

  @Override
  public Integer call() throws Exception {
    System.out.println(new Sample().getCustomerInfo(customerId));
    return 0;
  }
}

File: scalardb-dotnet-samples/scalardb-cluster-linq-sample/Commands/GetCustomerInfoCommand.cs
using System.CommandLine;

namespace ScalarDbClusterLinqSample.Commands;

public static class GetCustomerInfoCommand
{
    private const string Name = "GetCustomerInfo";
    private const string Description = "Get customer information";

    private const string ArgName = "id";
    private const string ArgDescription = "customer ID";

    public static Command Create()
    {
        var customerIdArg = new Argument<int>(ArgName, ArgDescription);
        var getCustomerInfoCommand = new Command(Name, Description)
        {
            customerIdArg
        };

        getCustomerInfoCommand.SetHandler(customerId =>
        {
            using var sample = new Sample();
            var customerInfo = sample.GetCustomerInfo(customerId);
            
            Console.WriteLine(customerInfo);
        }, customerIdArg);
        
        return getCustomerInfoCommand;
    }
}
